dotnet/src/Microsoft.AutoGen/Agents/InMemory/InMemoryAgentWorkerRuntime.cs

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// InMemoryAgentWorkerRuntime.cs
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using Microsoft.AutoGen.Abstractions;
+using Microsoft.AutoGen.Runtime;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AutoGen.Agents;
+
+public class InMemoryAgentWorkerRuntime : IAgentWorkerRuntime, IAgentWorkerRegistryGrain, IWorkerGateway, IAgentContext
+{
+    private readonly ILogger<InMemoryAgentWorkerRuntime> _logger;
+    private readonly InMemoryQueue<CloudEvent> _eventsQueue = new();
+    private readonly InMemoryQueue<Message> _messageQueue = new();
```

Comment:
The `InMemoryQueue<CloudEvent>` seems not doing anything more than `Channel<CloudEvent>`. Maybe just use `Channel<T>` instead?

Reply:
more coming here to ensure delivery semantics